### PR TITLE
Fix NodeToNodeVersion for Babbage and P2P.

### PR DESCRIPTION
**Pull request by coot**
- Use NodeToNodeV_9 for babbage and NodeToNodeV_10 for p2p
- In non P2P mode use InitiatorOnlyDiffusionMode in all handshake negotiations

### CODE REVIEW

**Review by amesgen — COMMENTED** [review]
FYI: We added `isPipeliningEnabled` here:

https://github.com/input-output-hk/ouroboros-network/blob/3c043a3acfdab0405198b1d3380610f0a55eb437/ouroboros-network/src/Ouroboros/Network/NodeToNode/Version.hs#L132-L135

The check for whether P2P is enabled could also be moved there, to centralize the feature checks by version.

**Comment by coot** [comment]
Thanks @amesgen, that's a good idea.

**Comment by coot** [comment]
Actually, I think that both `nodeDataFlow` and `localDataFlow` should belong to `Diffusion.P2P`. They are not exported, and there's no reason to make them available for downstream dependencies.

**Review by dcoutts — APPROVED** [review]
Sounds plausible to me.

**Comment by amesgen** [comment]
> Actually, I think that both `nodeDataFlow` and `localDataFlow` should belong to `Diffusion.P2P`. They are not exported, and there's no reason to make them available for downstream dependencies.

I wasn't necessarily suggesting to move these functions, but to add sth like `isP2PEnabled` next to `isPipeliningEnabled` and use that instead of `nv >=`-like checks.

**Comment by coot** [comment]
Yes, but we only do that in `nodeDataFlow`; putting it in `Ouroboros.Network.NodeToNode` would suggest that this is required by downstream dependencies or by other parts of the `ouroboros-network`.

**Comment by coot** [comment]
There was a CI [failure](https://hydra.iohk.io/build/15138118/nixlog/18), but it's fixed by PR #3772 (which fixes test itself, not the code).

**Comment by coot** [comment]
bors merge

**Review by dnadales — APPROVED** [review]

File: ouroboros-network-testing/src/Ouroboros/Network/Testing/Utils.hs [hunk 1]
```diff
@@ -5,7 +5,34 @@
...
+  , tracerWithTimeName
+  , swapTimeWithName
+  , swapNameWithTime
+  , splitWithNameTrace
+  -- * Tracers
+  , debugTracer
+  , sayTracer
+  -- * Tasty Utils
+  , nightlyTest
+  , ignoreTest
```
Comment (coot) [inline ouroboros-network-testing/src/Ouroboros/Network/Testing/Utils.hs:32]: The export list was added, to actually export `ignoreTest`.

**Comment by coot** [comment]
bors merge

**Comment by coot** [comment]
bors merge

**Review by kevinhammond — COMMENTED** [review]
, (NodeToNodeV_10, CardanoNodeToNodeVersion6)

Would be worth commenting to explain the two identical eras.

Otherwise looks good

**Pull request merged**